service/src/main/java/bio/terra/pipelines/common/utils/PipelineInputTypesEnum.java

```diff
@@ -71,16 +74,25 @@ public List<String> cast(String fieldName, Object value) {
   },
   VCF_ARRAY {
     @Override
-    public List<String> cast(String fieldName, Object value) {
+    public <T> T cast(String fieldName, Object value, TypeReference<T> typeReference) {
       validateNotNullOrEmpty(fieldName, value);
       if (value instanceof List<?> listValue) {
         try {
           List<String> stringList =
               listValue.stream()
-                  .map(itemValue -> VCF.cast(fieldName, itemValue).toString())
+                  .map(
+                      itemValue ->
+                          VCF.cast(fieldName, itemValue, new TypeReference<>() {}).toString())
```

Comment:
hm you still have `.toString()` here, and that's what we wanted to avoid - when implementing these changes, I still can't remove this

Reply:
ooh nevermind it works if the typereference input to that cast there is TypeReference<String> rather than TypeReference<>

Reply:
Oh whoops, just added a thing to make it behave